rtbot_plugin_manager/tests/test_plugin_manager.py

```diff
@@ -289,7 +289,7 @@ def test_load_syntactically_invalid_plugin(pm):
 def test_load_plugin_with_error_during_initialization(pm):
     with pytest.raises(PluginLoadingError) as exc:
         pm.load_plugin_from_string(PLUGIN_WITH_CONSTRUCTOR_EXCEPTION)
-    assert isinstance(exc.__cause__, RuntimeError)
+        assert isinstance(exc.__cause__, RuntimeError)
```

Comment:
Can you explain why this had to move into the `pytest.raises` context manager block? I don't actually see how it makes a difference here.

But as a rule of thumb, you want to keep your raise checks in unit tests as short as possible, because they always come at the risk of masking other errors.
To put it another way: when making an assertion that a certain block of code raises an exception, you want to be as specific as possible, ideally asserting that exactly that one call which is supposed to raise the exception in fact does.